src/com/selftest/queue/AbstractQueue.java

```diff
@@ -0,0 +1,56 @@
+
+package com.selftest.queue;
+
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+public abstract class AbstractQueue<T> implements Queue<T> {
+
+	protected T[] queue;
+	protected int putLoc, getLoc;
+
+	@SafeVarargs
+	public AbstractQueue( int size, T ... values ) {
+		queue = Arrays.copyOf( values, size );
+		putLoc = getLoc = 0;
+	}
+
+	@Override
+	public synchronized Iterator<T> iterator() {
+		return new Iterator<T>() {
+
+			int position = 0;
+
+			@Override
+			public boolean hasNext() {
+				return position < queue.length;

```

Comment:
Won't this iterate over a load of empty (or assumed deleted) indices by not considering `putloc` and `getloc`? Based on what this iterator does, you could have just returned: `return Arrays.asList(queue).iterator()`.